### PR TITLE
answers updated

### PR DESCRIPTION
**Pull request by shinheechae**

### CODE REVIEW

File: db-connect.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import mongoose from 'mongoose';
+
+ 
+
+//const Mongo_URL = "mongodb://localhost/users_db";
+const Mongo_URL = "mongodb+srv://shinhee:*******@shinhee.fglsw.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
```
Comment (losrobbos) [inline db-connect.js:6]: This connection string here will not work, if the password is missing. 
Or did you really choose ******** as password? :smile:

File: models/Student.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const StudentSchema = new Schema({
+    //! UNIQUE Constraint is actually the only rule stored in DB-wise!!
+    firstname: {type: String, required: true},
+    lastname: {type: String, required: true},
+    favQuote: {type: String},
+    isTutor: {type: Boolean, default: false},
+    role: {
+        type: String,
+        //  default: "user",
```
Comment (losrobbos) [inline models/Student.js:18]: This default there is actually correct. No need to comment it out :)

File: server.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+app.get("/students", async (req, res)=> {
+    const students = await Student.find()
+    res.json(students);
+
+}) 
+
+app.get("/students/migrate", async(req, res)=> {
+
+    const updated = await Student.updateMany().exists("role", false).set("role", "user")
```
Comment (losrobbos) [inline server.js:22]: Yesss! Perfect!

File: server.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        res.json(student.firstname)
+    } catch(err) {
+        next(err)
+    }
+})
+
+app.post('/students', async (req, res, next)=> {
+    try {
+        const body = req.body
+        const newbie = await Student.create(body)
```
Comment (losrobbos) [inline server.js:40]: Got that newbie!

File: server.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    } catch(err) {
+        next(err)
+    }
+})
+
+app.delete("/students/:id", async(req, res, next)=> {
+    try {
+        const {id} = req.params;
+        const studentDeleted = await Student.findByIdAndDelete(id);
+        res.json({done: `ID ${id} is deleted`});
```
Comment (losrobbos) [inline server.js:62]: Here you can actually return the deleted student, so `res.json( studentDeleted )`
Even though it is a bit confusing, findByIdAndDelete() will actually deliver back the deleted student. And the deleted item we would return back to the frontend, as a convention. The frontend can then decide what to do with it...

**Comment by losrobbos** [comment]
All good, Shinhee, very nice progress compared to your previous API :) You are well prepared for the next stages...

File: db-connect.js (hunk 1 above)
Comment (shinheechae) [reply db-connect.js:6]: thanks Robert for pointing this out. i now can use dotenv, so this won't happen again. :D

**Comment by shinheechae** [comment]
thank you for your feedbacks. :)

**Pull request closed without merging**